aws-codeguruprofiler-profilinggroup/src/test/java/software/amazon/codeguruprofiler/profilinggroup/UpdateHandlerTest.java

```diff
@@ -0,0 +1,267 @@
+                subject.handleRequest(proxy, request, null, logger);
+
+                verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(getPolicyRequest), any());
+                verify(proxy, times(1)).injectCredentialsAndInvokeV2(
+                    eq(RemovePermissionRequest.builder()
+                           .profilingGroupName(profilingGroupName)
+                           .actionGroup(ActionGroup.AGENT_PERMISSIONS)
+                           .revisionId(revisionId)
+                           .build()
+                    ),
+                    any());
+                verifyNoMoreInteractions(proxy);
+            }
+        }
+    }
+
+    @Nested
+    class WhenPermissionsIsProvided {
+        private final PutPermissionRequest putPermissionRequest =
+            PutPermissionRequest.builder()
+                .profilingGroupName(profilingGroupName)
+                .actionGroup(ActionGroup.AGENT_PERMISSIONS)
+                .principals(principals)
+                .build();
+
+        @BeforeEach
+        public void setup() {
+            request =  makeRequest(
+                ResourceModel.builder()
+                    .profilingGroupName(profilingGroupName)
+                    .permissions(
+                        Permissions.builder()
+                            .agentPermissions(
+                                AgentPermissions.builder()
+                                    .principals(principals)
+                                    .build()
+                            )
+                            .build()
+                    )
+                    .build()
+            );
+            doReturn(GetPolicyResponse.builder().build())
+                .when(proxy).injectCredentialsAndInvokeV2(eq(getPolicyRequest), any());
+            doReturn(PutPermissionResponse.builder().build())
+                .when(proxy).injectCredentialsAndInvokeV2(eq(putPermissionRequest), any());
+        }
+
+        @Test
+        public void testSuccess() {
+            final ProgressEvent<ResourceModel, CallbackContext> response
+                = subject.handleRequest(proxy, request, null, logger);
+
+            assertThat(response).isNotNull();
+            assertThat(response.getStatus()).isEqualTo(OperationStatus.SUCCESS);
+            assertThat(response.getCallbackContext()).isNull();
+            assertThat(response.getCallbackDelaySeconds()).isEqualTo(0);
+            assertThat(response.getResourceModel()).isEqualTo(request.getDesiredResourceState());
+            assertThat(response.getMessage()).isNull();
+            assertThat(response.getErrorCode()).isNull();
+        }
+
+        @Test
+        public void testCorrectOperationsAreCalled() {
+            subject.handleRequest(proxy, request, null, logger);
+
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(getPolicyRequest), any());
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(putPermissionRequest), any());
```

Comment:
Can we validate this one is called with the correct arguments?

Reply:
Oops, we already do -- `eq(putPermissionRequest)`, the second argument is not as important. Please ignore